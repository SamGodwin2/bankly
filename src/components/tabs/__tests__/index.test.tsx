import * as React from 'react'
import TestRenderer from 'react-test-renderer'
import ShallowRenderer from 'react-test-renderer/shallow'
import Tabs from '../';
import Tab from '../tab';

import type { ReactTestRendererJSON } from 'react-test-renderer';

const renderer = ShallowRenderer.createRenderer()

describe('Account component', () => {
  test('should render and match the snapshot', () => {
    const mockOnChange = jest.fn()

    renderer.render(
      <Tabs activeTab='foo' onChange={mockOnChange}>
        <Tab value='foo'>Foo</Tab>
        <Tab value='bar'>Bar</Tab>
      </Tabs>
    )
    const output = renderer.getRenderOutput()

    expect(output).toMatchSnapshot();
  });

  test('should render as an unordered list', function () {
    const mockOnChange = jest.fn()

    const testRenderer = TestRenderer.create(
      <Tabs activeTab='foo' onChange={mockOnChange}>
        <Tab value='foo'>Foo</Tab>
        <Tab value='bar'>Bar</Tab>
      </Tabs>,
    )
    const output = testRenderer.toJSON() as ReactTestRendererJSON

    expect(output.type).toBe('ul')
  })
});
