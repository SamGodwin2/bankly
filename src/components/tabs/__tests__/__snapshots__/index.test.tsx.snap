// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account component should render and match the snapshot 1`] = `
<nav>
  <Tab
    isActive={true}
    onClick={[MockFunction]}
    value="foo"
  >
    Foo
  </Tab>
  <Tab
    isActive={false}
    onClick={[MockFunction]}
    value="bar"
  >
    Bar
  </Tab>
</nav>
`;
